### DISCUSSION TITLE
BUG: General Users see All Customers and Projects

### DISCUSSION THREAD

**Discussion by xstable** [kimai/kimai#4161] [General]
Is it default setting that a user (no Teamlead), can see all Customers and their projects in the `create timeentries` section?

![image](https://github.com/kimai/kimai/assets/1778068/d77599c6-6e5b-4194-9650-752cc52a9c0b)

I think the user should only see Projects which he is assigned to!
If I open his Profile-Page, I only see this one Project: 

![image](https://github.com/kimai/kimai/assets/1778068/262f1b5b-7553-438f-b83d-74768e38597b)

The User where I've made the screenshot is only assigned to Project "TyrePortal", but he can see all other Customers and Projects there. 
Is this a bug, or did I missconfigure something (btw: I didn't changed the roles yet)

**Comment by kevinpapst** [comment]
Read permission docs. All projects without team assignment are globally visible.

**Reply by xstable** [reply]
Oh, that's good to know, thanks. 

Generally (in most software) it's the case, that nobody see anything he isn't allowed/assigned to... why do you choose other way around?
Is there any use case for such an "generally world open, restricted by need" Philosophie? *(This is not a criticism, I want to understand the reason / use-case)*

**Reply by kevinpapst** [reply]
> in most software

I would love to see this comparison chart...

> Is there any use case

No, this was blindly stupid by me 👿 

> I want to understand the reason / use-case

No, I don't believe that. You are questioning the maintainers decisions.

Many companies do not require team limitations. That's why.

**Comment by xstable** [comment]
Oh that's strange. 
I did following: 

1. Login as Admin, 
2. create "AdminOnyl" team.
3. assigned this team to the first 3 projects 
4. LogOut and re-login as user. 
5. he now ONLY see his UIM-Customer and ONLY his Project (that's how I would expected it in first case)

Was confused about this, because therewhere many other projects before, that where listed there in the Pulldown, and now they are gone. 
To verify if my "OnlyAdmin" team was the cause I did following:
1. re-login as admin
2. un-assign those three Projects from "admin-only" team
3. re-login as user

Result, Select-Box still show only the project he is assigned to: 

![image](https://github.com/kimai/kimai/assets/1778068/0fa16e01-734d-498a-90ba-08024fea3451)

I don't see why that happend, and I don't know what the right behavior is. Can you clarify this for me?

**Reply by kevinpapst** [reply]
https://www.kimai.org/documentation/teams.html

> Customers - access can be limited by assigning one or more teams. Customers without any team assignment can be seen by every user. **Projects for this customer will be limited as well, if a team is assigned**

**Reply by xstable** [reply]
ok, I see. But that is not what my setup currently do. 
Wait ... I'll show you... 

![image](https://github.com/kimai/kimai/assets/1778068/30ce0822-a408-4c8d-8577-7176ed8d496a)

Based on your explanation, first 5 projects in the screenshot above and the 7th should be visible by all users, right... (as they have no team assigned at all)
... and that is what I've posted in my initial posting... there you see all the other projects. 

BUT now it's not that way anymore, after running trought the steps I've send in my last post. 
Look here, the select-box from general User-Role perspective (at 2. choosed Project-overview): 
![image](https://github.com/kimai/kimai/assets/1778068/4f1e556e-7cf1-422a-9122-00de5fbce940)

Same here on Time-Tracker: 
![image](https://github.com/kimai/kimai/assets/1778068/1ff15c4a-ef82-4aa4-9633-a0980d126554)

I like it that way much more as described in my initial post. But I'm afraid that one of both behaviors are now cause of bug.

**Reply by xstable** [reply]
BTW: If you like we can do a short zoom-meeting (tomorrow) and you can have a look at it.

**Reply by kevinpapst** [reply]
Are you a customer?

**Reply by xstable** [reply]
Not sure what you mean. Isn't it, that Customers in Backend don't have an login (I don't know, I might be wrong) 
As far as I can see the "Customers" Section only contain "Companies" which can have Projects assigned, but I didn't seen anything that they have own logins. Or do you mean something else?

In the above case, UIM is the Customer, and Alex is the user who track times on the Project "TyrePortal" of Customer UIM.

**Comment by xstable** [comment]
Work as expected in 2.0
